docs/design/storagenode-graceful-exit.md

```diff
@@ -0,0 +1,62 @@
+
+
+## GOALS
+
+- Give Storage Nodes a mechanism to leave the network while receiving their escrows, and reduce repair caused by node churn on satellites. 
+
+
+## NON GOALS
+- Sending the storage nodes escrows
+	- The sending of tokens will happen through our normal token payment process
+
+
+##SCENARIOS
+
+- A storage node successfully completes a graceful exit
+- A storage node cancels the graceful exit process
+- A storage node shuts down during the graceful exit process
+	- Power loss or reboot 
+- A storage node runs out of bandwidth during the graceful exit process
+- A storage node wants to “partially” exit a satellite
+- A storage node wants to rejoin a satellite it previously exited
+
+
+##Business Requirements/ Job Stories
+
+- When a Storage Node runs the graceful exit command I want them to be prompted with a confirmation message so that we can avoid storage nodes running the command accidentally
+	- Something like “Are you sure you want to run graceful exit? Graceful exit sends the 1.2 TB of data you are storing to other storage nodes so you will no longer be paid for storing this data. This process can take several hours to complete please your storage node running until it is completed.”
+	- We need to specify the amount of data the Storage node will be sending to other nodes. 
+- When a Storage Node completes graceful exit for all satellites I want their Node to automatically shut down so that they do not have to worry about receiving any more data.
+- When a Storage Node completes graceful exit I want all of the data they have been storing to be deleted so that their computer is no longer storing any pieces. 
+- When a Storage node no longer wants to store data for a satellite I want them to have the ability to run graceful exit for a specific satellite so that they do not lose their escrow on that satellite. 
+	- Add the ability to run graceful exit for a specific satellite
+	- The satellite must mark the storage node as ‘exited’ and the SN should update its satellite whitelist so that it rejects any future requests from that satellite.
+- When a Storage Node operator wants to reduce the amount of storage space they have allocated to the network I want them to have the ability to do a “partial graceful exit which would transfer some of the data they currently have onto other nodes so they have a way of reducing thier storage allocation without just deleting data and failing audits. 
```

Comment:
When a Storage Node operator has dropped data and wants to avoid audit penalty I want them to have the ability to do a "partial graceful exit" which would remove the piece from the pointerDB.

Reply:
Good point! We need gracefully exiting nodes to proof that they transferred their pieces successfully to the receiving nodes and the transferred data was the correct one. At some point we were discussing if we should store a hash for each piece in the pointer db, but then we decided to drop it to save space. If we had this hash in the pointer db then the receiving node can send its hash to the satellite, and the satellite can verify it with the one it has in the pointer db.

Reply:
yea I agree, we need to figure out a strategy for this. perhaps we should make it so that the storage node does not get to choose which pieces it sends to other nodes during a partial graceful exit. The reason I added that requirement was because Helene brought up a good point which was what if a SNO wants to reduce how much storage space the allocate to the network. @littleskunk @kaloyan-raev does that prevent cheating?

Reply:
Regardless if the graceful exit is full or partial, we need to think about what happens if the exiting node cannot proof the successful transfer of all pieces? These might be exactly the pieces that a cheater node is missing and is trying to escape the potential disqualification by attempting a graceful exit.

Should we mark the graceful exit as unsuccessful and disqualify the node if a certain percentage of piece transfers are not successful?